client/components/product.js

```diff
@@ -24,22 +27,34 @@ export class productComponent extends Component {
           <p>Description: {product && product.description}</p>
           <form onSubmit={async (event) => {
             event.persist();
-            let { cartId, user } = this.props;
             if (Object.keys(user).length === 0 && user.constructor === Object) {
-              const respond = await this.props.handleCreateGuest(event);
-              user = respond.user;
+              const response = await this.props.handleCreateGuest(event);
+              user = { response };
```

Comment:
I think you changed the meaning of Thomas' code. I think we may need to disable the rule for this file. I was thinking maybe a ternary would be best for this. I'll look up ternary with multiple statements.

Reply:
So ternary can be used with multiple statements. But yeah we should just disable the rule.

Also regarding the `Object.keys` logic, Omri said to put that in a helper function.

Reply:
`/* eslint-disable prefer-destructuring */`

Reply:
`({ user } = response);`